### PR TITLE
Example of safe type casting

### PR DESCRIPTION
**Pull request by Morheit**

### CODE REVIEW

File: src/DBConnection2.hpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ class MySqlDBConnection2 : public IDBConnection2 {
 
 class SqLiteDBConnection2 : public IDBConnection2 {
 public:
-    SqLiteDBConnection2() = default;
+    SqLiteDBConnection2() : m_protocol_version(0) {};
```
[the remark this answers is not in the document]
Comment (Morheit) [reply src/DBConnection2.hpp]: @malirod , you're right. Removed

File: src/DBConnection.hpp [hunk 2]
```diff
@@ -4,12 +4,12 @@
 #include <utility>
 
 struct Info {
-    Info() = default;
+    Info() : server_version{""}, protocol_version(0) {};
```
[the remark this answers is not in the document]
Comment (Morheit) [reply src/DBConnection.hpp]: @malirod , no. Removed

File: CMakeLists.txt [hunk 3]
```diff
@@ -12,9 +12,21 @@ find_package(Threads REQUIRED)
...
-set(SRC_LIST "src/DBConnection.cpp" "src/DBConnection.hpp" src/DBConnection2.cpp src/DBConnection2.hpp src/ConnectionDispatcher.cpp src/ConnectionDispatcher.hpp)
-
-add_executable(testrunner "${SRC_LIST}" test/ConnectionTest.cpp test/ConnectionTest2.cpp)
+set(SRC_LIST
+    "src/DBConnection.cpp"
+    "src/DBConnection.hpp"
+    "src/DBConnection2.cpp"
+    "src/DBConnection2.hpp"
+    "src/ConnectionDispatcher.cpp"
+    "src/ConnectionDispatcher.hpp"
```
[the remark this answers is not in the document]
Comment (Morheit) [reply CMakeLists.txt]: @malirod , removed

File: test/ConnectionTest4.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    std::string getMySqlConnection4AdvancedInfo(IDBConnection4& dbConnection) {
+        if (ConnectionType::MYSQL == dbConnection.connectionType()) {
+            return static_cast<MySqlDBConnection4&>(dbConnection).advancedQuery().server_version;
+        }
+        return dynamic_cast<MySqlDBConnection4&>(dbConnection).advancedQuery().server_version;
+    }
+
+    int getSqLiteConnection4Info(IDBConnection4& dbConnection) {
```
[the remark this answers is not in the document]
Comment (Morheit) [reply test/ConnectionTest4.cpp]: @malirod , done

File: test/ConnectionTest4.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
+#include <gtest/gtest.h>
+
+#include "DBConnection4.hpp"
+
+namespace {
+
+    void checkMySqlConnection4Info(IDBConnection4& db_connection, const int expected_value) {
+        if (db_connection.connectionType() == ConnectionType::MYSQL) {
```
Comment (malirod) [inline test/ConnectionTest4.cpp]: Try to keep structure of is statements simpler. Also try to follow next idea, make asserts and checks first and continue if all fine
I suggest to rewrite this way
'''
If not mysqls then throw;
Assert cast
'''
Update here and bow

Comment (Morheit) [reply test/ConnectionTest4.cpp]: @malirod , done

**Pull request merged**